src/utils/generator.js

```diff
@@ -292,6 +299,9 @@ class PossiblyNormalReturnEntry extends GeneratorEntry {
     let alternateEffects =
       completion.alternate instanceof AbruptCompletion ? completion.alternateEffects : empty_effects;
     this.alternateGenerator = Generator.fromEffects(alternateEffects, realm, "AlternateEffects");
+
+    invariant(this.consequentGenerator.isTerminal() === this.alternateGenerator.isTerminal());
+    this.isTerminal = this.consequentGenerator.isTerminal();
```

Comment:
Only assign if true.

Reply:
It's not optional in `GeneratorEntry`.

Reply:
A PossiblyNormalCompletion entry never terminates a generator.